core/src/services/gcs/backend.rs

```diff
@@ -520,6 +521,37 @@ impl Accessor for GcsBackend {
             GcsPager::new(self.core.clone(), path, "", args.limit(), None),
         ))
     }
+
+    async fn presign(&self, path: &str, args: OpPresign) -> Result<RpPresign> {
+        // We will not send this request out, just for signing.
+        let mut req = match args.operation() {
+            PresignOperation::Stat(v) => {
+                self.core
+                    .gcs_head_object_request(path, v.if_match(), v.if_none_match())?
```

Comment:
Sadly, gcs only supports query sign over XML API: https://cloud.google.com/storage/docs/request-endpoints#xml-api

So we can't reuse `gcs_head_object_request` and `gcs_insert_object_request` here.

For more information, please visit: https://cloud.google.com/storage/docs/access-control/signed-urls